cloudinit/ssh_util.py

```diff
@@ -544,11 +544,27 @@ def parse_ssh_config_map(fname):
     return ret
 
 
+def _includes_dconf(fname: str) -> bool:
+    if not os.path.isfile(fname):
+        return False
+    with open(fname, "r") as f:
+        for line in f:
+            if f"Include {fname}.d/*.conf" in line:
```

Comment:
- this also matches a commented Include statement (eg `# Include /etc/ssh/sshd_config.d/*.conf`)
- why does the include name need to match `fname` ? there could also be eg `Include /etc/foobar.d/`

Reply:
* Fixed, thanks.
* To make it future-proof. To make it compatible with the case where other distros or ssh server implementations could locate the sshd_config in other place but following the same pattern (include {fname}.d directory). [update_ssh_config](https://github.com/canonical/cloud-init/pull/1618/files#diff-8978d79f04e525de3011b92f7b141a7bd6dae4b6d0a70f9b9ea923bbd1451a43R557) receives fname as arg defaulting to `/etc/ssh/sshd_config` and we only use the default at the moment, but that could change in the future.